ci/ci/build.py

```diff
@@ -680,11 +680,12 @@ def build(self, batch, code, scope):
 '''
                 elif w['kind'] == 'Service':
                     assert w['for'] == 'alive', w['for']
+					resource_type = w.get("resource_type", "deployment")
                     port = w.get('port', 80)
                     timeout = w.get('timeout', 60)
                     script += f'''
 set +e
-kubectl -n {self.namespace} rollout status --timeout=1h deployment {name} && \
+kubectl -n {self.namespace} rollout status --timeout=1h {resource_type} {name} && \
   kubectl -n {self.namespace} wait --timeout=1h --for=condition=available deployment {name} && \
```

Comment:
needed here too

Reply:
Unfortunately, I don't think this will uniformly work. 

https://github.com/kubernetes/kubernetes/issues/79606

The rollout command works fine, but when I wait --for=condition=available on grafana, it will never find the condition met.

Reply:
This may be fixed in 1.12 (https://github.com/kubernetes/kubernetes/pull/62943), and we're in 1.14. Helm people still seem to be having issues: https://github.com/helm/helm/issues/4539